pkg/kube/inject/inject.go

```diff
@@ -889,3 +898,61 @@ func GetProxyIDs(namespace *corev1.Namespace) (uid int64, gid int64) {
 
 	return
 }
+
+// GetProxyIDsFromPod returns the UID and GID to be used in the RunAsUser and RunAsGroup fields in the template
+// It relies on OpenShift injecting a valid UID into the pod template which we can then reuse.
+func GetProxyIDsFromPod(pod *corev1.Pod) (uid int64, gid int64) {
+	var proxyUID, proxyGID *int64
+	tproxyInterceptionMode := pod.Annotations != nil && pod.Annotations["sidecar.istio.io/interceptionMode"] == string(model.InterceptionTproxy)
```

Comment:
nit: Use the constant `annotation.SidecarInterceptionMode.Name` instead of hardcoding it.

`import "istio.io/api/annotation"`